src/mfuns2.c

```diff
@@ -3460,3 +3460,37 @@ mfn_width(MFUNARGS)
 
     return buf;
 }
+
+/**
+ * MPI function that returns if an object if it has the given flag (re)set.
+ * 
+ * @see has_flag
+ * 
+ * @param descr the descriptor of the caller
+ * @param player the ref of the calling player
+ * @param what the dbref of the trigger
+ * @param perms the dbref for permission consideration
+ * @param argc the number of arguments
+ * @param argv the array of strings for arguments
+ * @param buf the working buffer
+ * @param buflen the size of the buffer
+ * @param mesgtyp the type of the message   
+ * @return string parsed results
+ */
+
+const char *
+mfn_flagp(MFUNARGS)
+{
+    dbref obj = mesg_dbref_local(descr, player, what, perms, argv[0], mesgtyp);
+
+    if (obj == PERMDENIED || obj == AMBIGUOUS || obj == UNKNOWN || obj == NOTHING ||
```

Comment:
It looks like a "failed match" error is the more common way to handle this.  Here's a sample I found in another primitive:

```
if (obj == AMBIGUOUS || obj == UNKNOWN || obj == NOTHING || obj == HOME) {
        ABORT_MPI("FLAG?", "Failed match. (arg1)");
}
```

Reply:
OH ... and obj==PERMDENIED in that list, too. Whoops :)  Basically, keep your if statement, but change the return to ABORT_MPI